cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -121,26 +122,29 @@ LocalTrajectoryBuilder::AddRangeData(const common::Time time,
 
   if (num_accumulated_ >= options_.scans_per_accumulation()) {
     num_accumulated_ = 0;
+    const transform::Rigid3d gravity_alignment = transform::Rigid3d::Rotation(
+        extrapolator_->EstimateGravityOrientation(time));
     return AddAccumulatedRangeData(
-        time, sensor::TransformRangeData(accumulated_range_data_,
-                                         tracking_delta.inverse()));
+        time,
+        // Transforms 'accumulated_range_data_' into a frame where gravity
+        // direction is approximately +z.
+        TransformToGravityAlignedFrameAndFilter(
+            gravity_alignment.cast<float>() * tracking_delta.inverse(),
+            accumulated_range_data_));
   }
   return nullptr;
 }
 
 std::unique_ptr<LocalTrajectoryBuilder::MatchingResult>
 LocalTrajectoryBuilder::AddAccumulatedRangeData(
-    const common::Time time, const sensor::RangeData& range_data) {
-  // Transforms 'range_data' from the tracking frame into a frame where gravity
-  // direction is approximately +z.
-  const transform::Rigid3d gravity_alignment = transform::Rigid3d::Rotation(
-      extrapolator_->EstimateGravityOrientation(time));
-  const sensor::RangeData gravity_aligned_range_data =
-      TransformAndFilterRangeData(gravity_alignment.cast<float>(), range_data);
+    const common::Time time,
+    const sensor::RangeData& gravity_aligned_range_data) {
   if (gravity_aligned_range_data.returns.empty()) {
     LOG(WARNING) << "Dropped empty horizontal range data.";
     return nullptr;
   }
+  const transform::Rigid3d gravity_alignment = transform::Rigid3d::Rotation(
```

Comment:
Can we pass this as an argument? It is closely tied to the `time` and `gravity_aligned_range_data` we take as arguments.